pkg/gameservers/gameservers.go

```diff
@@ -72,14 +72,19 @@ func address(node *corev1.Node) (string, error) {
 
 // applyGameServerAddressAndPort gathers the address and port details from the node and pod
 // and applies them to the GameServer that is passed in, and returns it.
-func applyGameServerAddressAndPort(gs *agonesv1.GameServer, node *corev1.Node, pod *corev1.Pod) (*agonesv1.GameServer, error) {
+func applyGameServerAddressAndPort(gs *agonesv1.GameServer, node *corev1.Node, pod *corev1.Pod, syncPodPortsToGameServer func(*agonesv1.GameServer, *corev1.Pod) error) (*agonesv1.GameServer, error) {
```

Comment:
🤔  curious why you are passing in the function, rather than passing through a `CloudProduct` interface? Easier testing?

Reply:
It was mostly for easier testing, yeah - also a general desire to pass as narrow interface as possible (I didn't create the interface for this because it's a single function).